packages/node_modules/@node-red/runtime/lib/flows/Flow.js

```diff
@@ -128,6 +128,65 @@ class Flow {
         }
         this.parent.log(msg);
     }
+	
+    /**
+     * Checks if node A and node B are in the same group. 
+     * Node B can also be placed in a subgroup.
+     * If node A is not in any group, false is returned
+     * @param {Node} nodeIdA   Node which defines the first search level
+     * @param {Node} nodeIdB   Node which is to be searched in the group or a subgroup
+     * @returns {boolean}      Returns true if all nodes are in the same group. If not, then false or if node A is not in a group then also false.
+     */
+    isNodeInSameGroup(nodeIdA, nodeIdB) {
```

Comment:
This code iterates over all groups in the full configuration; I think there is a more efficient way to check for this. We also now have the scenario where you may have nested groups so you need to find the "closest" node.

I think if we update the `Node` object to have knowledge of its parent group id, we can simplify things a lot.


Lets use the following scenario:

```

 ┌─────────────────────────────────────┐
 │GroupA                               │
 │     ┌───────────────────────────┐   │
 │     │GroupB                     │   │
 │     │          ┌──────┐         │   │
 │     │          │NodeB │         │   │
 │     │          └──────┘         │   │
 │     │          ┌──────┐         │   │
 │     │          │CatchB│         │   │
 │     │          └──────┘         │   │
 │     └───────────────────────────┘   │
 │                ┌──────┐             │
 │                │CatchA│             │
 │                └──────┘             │
 └─────────────────────────────────────┘
```

To check if `CatchA` can 'see' `NodeB`, first check to see if their `g` properties are the same - if so, they are in the same group, return `0`.
Otherwise, walk up the parent groups of `NodeB`, at each level check to see if it is the group containing `CatchA` until we have no more parent groups. Return a count of how many parent levels were walked up, otherwise return `-1` to indicate no match.

The returned count can then be used as a 'distance' between the nodes; for any candidate node with a distance `>=0` find the one with the smallest distance and use that.

So with the example above, `CatchB` would have a distance of `0` and `CatchA` a distance of `1` - so we'd pick `CatchB`.

I'm going to work on this alternative implementation in a separate branch based on what you've done here.